### PR TITLE
simplewallet: reject invalid argument for boolean parameter

### PR DESCRIPTION
**Pull request by stoffu**
Addresses #2648

### CODE REVIEW

**Comment by moneromooo-monero** [comment]
I guess it's ok, though it feels like it could be compressed/factored, that's a lot of boilerplate.

**Comment by stoffu** [comment]
Thanks for your suggestion, I totally agree. Now refactored.

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -161,20 +161,49 @@ namespace
...
+    if (boost::algorithm::equals("false", s, ignore_case) || boost::algorithm::equals(simple_wallet::tr("false"), s, ignore_case))
+    {
+      result = false;
       return true;
+    }
 
     return false;
   }
 
+  bool parse_bool_and_use(const std::string& s, const std::function<void(bool)>& func)
```
Comment (vtnerd) [inline src/simplewallet/simplewallet.cpp]: Minor point - I would make this:
```c++
template<typename F>
bool parse_bool_and_use(const std::string& s, F func) {
    ...
}
```

**Comment by stoffu** [comment]
@vtnerd Thanks for commenting. Updated.

**Pull request merged**